addons/survey/tests/common.py

```diff
@@ -314,3 +314,80 @@ def setUp(self):
             'sequence': 3,
             'question_type': 'numerical_box',
         })
+
+
+class TestSurveyResultsCommon(SurveyCase):
+    def setUp(self):
```

Comment:
Can you move everything into a setupClass ? It's better as only called once for all tests.

Reply:
Did you try modifying "SurveyCase" and change the setup to a setupClass?

It would be "cls._type_match" instead of "self._type_match" but the rest should work.

But no you cannot call setup in setupClass.
The logic is that setUpClass is called only once for all the tests of that class.
And setUp is called once **before every single** test of that class.
Generally using setupClass is better as it creates everything once, and then the savepoint mechanism of the TransactionCase will "reset" the dataset for each test of the class, meaning it will end up being the same result as if it was done in setUp in terms of data, but it's faster.